contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -491,6 +506,64 @@ contract SellerHandlerFacet is SellerBase {
         return (pl.cloneAddress[_sellerId], pl.additionalCollections[_sellerId]);
     }
 
+    /**
+     * @notice Returns the availability of salt for a seller.
+     *
+     * @param _adminAddres - the admin address to check
+     * @param _salt - the salt to check (corresponds to `collectionSalt` when `createSeler` or `createNewCollection` is called or `newSalt` when `updateSellerSalt` is called)
+     * @return isAvailable - salt can be used
+     */
+    function isSellerSaltAvailable(address _adminAddres, bytes32 _salt) external view returns (bool isAvailable) {
+        bytes32 sellerSalt = keccak256(abi.encodePacked(_adminAddres, _salt));
+        return !protocolLookups().isUsedSellerSalt[sellerSalt];
+    }
+
+    /**
+     * @notice Calculates the expected collection address and tells if it's still avaialble.
+     *
+     * @param _sellerId - the seller id
+     * @param _collectionSalt - the collection specific salt
+     * @return collectionAddress - the collection address
+     * @return isAvailable - whether the collection address is available
+     */
+    function calculateCollectionAddress(
+        uint256 _sellerId,
+        bytes32 _collectionSalt
+    ) external view returns (address collectionAddress, bool isAvailable) {
+        (bool exist, Seller storage seller, AuthToken storage authToken) = fetchSeller(_sellerId);
+        if (!exist) {
+            return (address(0), false);
+        }
+
+        // get seller salt
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        bytes32 sellerSalt = lookups.sellerSalt[_sellerId];
+
+        // If seller salt is not set, calculate it
+        if (sellerSalt == 0) {
+            address admin = seller.admin;
+            if (admin == address(0)) {
+                admin = IERC721(lookups.authTokenContracts[authToken.tokenType]).ownerOf(authToken.tokenId);
+            }
+            sellerSalt = keccak256(abi.encodePacked(admin, _collectionSalt));
+        }
+
+        // Calculate collection address
+        bytes32 collectionSalt = keccak256(abi.encodePacked(sellerSalt, _collectionSalt));
+        bytes32 bytecodeHash = keccak256(
+            abi.encodePacked(
```

Comment:
@zajck I'd like you to explain this

Reply:
did I find the key here: https://blog.openzeppelin.com/deep-dive-into-the-minimal-proxy-contract ?

Reply:
This is directly connected to how all voucher clones are created. Lines 554-558 here represent the minimal clone bytecode that is deployed for each collection, equivalent to how we construct the bytecode in `cloneBosonVoucher`:

https://github.com/bosonprotocol/boson-protocol-contracts/blob/e6d14e4f4dfe8089f4276389a7725023672116a8/contracts/protocol/bases/SellerBase.sol#L177-L188

Then it just needs to be hashed in order to compute create2 address as per the definition.